src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    protected a: number,
+    protected b: number,
+    protected c: number,
+  ) {
+    if ((a >= b + c) || (b >= c + a) || (c >= a + b)) {
+      throw new Error('The length side is less than sum of both');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The length side is zero or less');
+    }
+  }
 
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+
+    return (Math.round(Math.sqrt(p * (p - this.a) * (p - this.b)
+    * (p - this.c)) * 100) / 100);
+  }
```

Comment:
Make it more readable
Also, you can not give a type to variables if you put some value to this variable
Typescript knows about this type
But it's up to you
```suggestion
    const semiPerimeter  = (this.a + this.b + this.c) / 2;
    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c);

    return Math.round(area) * 100) / 100);
  }
```